Makefile

```diff
@@ -311,6 +319,34 @@ CFLAGS     += $(ARCH_FLAGS) \
               -MMD -MP \
               $(EXTRA_FLAGS)
 
+# MacOS arm64 specific flags
+ifeq ($(shell uname -s),Darwin)
```

Comment:
See above; this can be implemented as combination of ARCH_FLAGS / ARCH_CFLAGS_REMOVE

And it must be specific for SITL, otherwise it will affect building of normal targets